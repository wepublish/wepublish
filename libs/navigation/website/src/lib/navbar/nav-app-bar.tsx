import {MdAccountCircle, MdClose, MdMenu, MdOutlinePayments} from 'react-icons/md'
import {AppBar, Box, css, styled, Toolbar, useTheme} from '@mui/material'
import {useMemo} from 'react'
import {BuilderNavAppBarProps, useWebsiteBuilder} from '@wepublish/website/builder'
import {navigationLinkToUrl} from '../link-to-url'
import {UseToggle} from '@wepublish/ui'
import {useUser} from '@wepublish/authentication/website'
import {FullNavigationFragment} from '@wepublish/website/api'

export const NavAppBar = ({
  logo,
  loginUrl,
  profileUrl,
  subscriptionsUrl,
  headerItems,
  menuToggle,
  actions
}: BuilderNavAppBarProps) => {
  const appBarStyles = useAppBarStyles(menuToggle.value)
  return (
    <AppBar position="static" elevation={0} color={'transparent'} css={appBarStyles}>
      <NavbarInnerWrapper>
        <NavbarMain>
          <NavbarIconButtonWrapper>
            <NavbarOpenCloseButton toggle={menuToggle} />
          </NavbarIconButtonWrapper>
          {!!headerItems?.links.length && (
            <NavbarLinks isMenuOpen={menuToggle.value}>
              <MenuItems items={headerItems} />
            </NavbarLinks>
          )}
        </NavbarMain>

        <HomeLogoButton logo={logo} menuToggle={menuToggle} />
        <NavbarActions isMenuOpen={menuToggle.value}>
          {actions}
          <LoggedInButtons profileUrl={profileUrl} subscriptionsUrl={subscriptionsUrl} />
          <LoggedOutButtons loginUrl={loginUrl} />
        </NavbarActions>
      </NavbarInnerWrapper>
    </AppBar>
  )
}

type HomeLogoButton = Pick<BuilderNavAppBarProps, 'logo' | 'menuToggle'> & {className?: string}

export const HomeLogoButton = ({logo, menuToggle, className}: HomeLogoButton) => {
  const logoLinkStyles = useLogoLinkStyles(menuToggle.value)
  const imageStyles = useImageStyles()
  const {
    elements: {Link, Image}
  } = useWebsiteBuilder()
  return (
    <Link href="/" aria-label="Startseite" css={logoLinkStyles} className={className}>
      <NavbarLogoWrapper>
        {!!logo && <Image image={logo} css={imageStyles} loading="eager" fetchPriority="high" />}
      </NavbarLogoWrapper>
    </Link>
  )
}

type LoggedInButtonsProps = Pick<BuilderNavAppBarProps, 'subscriptionsUrl' | 'profileUrl'>

export const LoggedInButtons = ({subscriptionsUrl, profileUrl}: LoggedInButtonsProps) => {
  const {hasUser} = useUser()
  const {
    elements: {Link, IconButton}
  } = useWebsiteBuilder()

  if (!hasUser) {
    return <></>
  }
  return (
    <>
      {subscriptionsUrl && (
        <Link href={subscriptionsUrl}>
          <IconButton css={{fontSize: '2em', color: 'black'}}>
            <MdOutlinePayments aria-label={'Subscriptions'} />
          </IconButton>
        </Link>
      )}

      <Link href={profileUrl}>
        <IconButton className="login-button" css={{fontSize: '2em', color: 'black'}}>
          <MdAccountCircle aria-label="Profil" />
        </IconButton>
      </Link>
    </>
  )
}

type LoggedOutButtonsProps = Pick<BuilderNavAppBarProps, 'loginUrl'>

export const LoggedOutButtons = ({loginUrl}: LoggedOutButtonsProps) => {
  const {hasUser} = useUser()
  const {
    elements: {Link, IconButton}
  } = useWebsiteBuilder()

  if (hasUser) {
    return <></>
  }
  return (
    <>
      <Link href={loginUrl}>
        <IconButton className="login-button" css={{fontSize: '2em', color: 'black'}}>
          <MdAccountCircle aria-label="Login" />
        </IconButton>
      </Link>
    </>
  )
}

type MenuItemsProps = {
  items: FullNavigationFragment | null | undefined
}

export const MenuItems = ({items}: MenuItemsProps) => {
  const {
    elements: {Link}
  } = useWebsiteBuilder()
  return (
    <>
      {items?.links.map((link, index) => (
        <Link key={index} href={navigationLinkToUrl(link)}>
          {link.label}
        </Link>
      ))}
    </>
  )
}

export const useAppBarStyles = (isMenuOpen: boolean) => {
  const theme = useTheme()

  return useMemo(
    () =>
      isMenuOpen
        ? css`
            background-color: ${theme.palette.primary.main};
            color: ${theme.palette.primary.contrastText};
          `
        : null,
    [theme, isMenuOpen]
  )
}

export const NavbarInnerWrapper = styled(Toolbar)`
  display: grid;
  grid-template-columns: max-content max-content 1fr;
  align-items: center;
  grid-auto-flow: column;
  justify-items: center;
  min-height: unset;
  padding: 0;
`

export const NavbarLinks = styled('div')<{isMenuOpen?: boolean}>`
  display: none;
  gap: ${({theme}) => theme.spacing(2)};
  align-items: center;
  justify-content: flex-start;
  width: 100%;
  ${({isMenuOpen}) =>
    isMenuOpen &&
    css`
      z-index: -1;
    `} @media (
  min-width: 740px) {
    // custom for maximum space usage
    display: flex;
    a {
      font-size: 1rem;
      text-decoration: none;
      color: ${({theme}) => theme.palette.common.black};
      ${({theme}) => theme.breakpoints.up('md')} {
        font-size: 1.3rem;
      }
    }
  }
`

export const NavbarMain = styled('div')<{isMenuOpen?: boolean}>`
  display: grid;
  grid-template-columns: max-content 1fr;
  align-items: center;
  justify-self: start;
  gap: ${({theme}) => theme.spacing(2)};
  ${({isMenuOpen}) =>
    isMenuOpen &&
    css`
      z-index: -1;
    `}
`

export const NavbarActions = styled(Box)<{isMenuOpen?: boolean}>`
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-self: end;
  gap: ${({theme}) => theme.spacing(1)};
  padding-right: ${({theme}) => theme.spacing(1)};
  justify-self: end;
  ${({isMenuOpen}) =>
    isMenuOpen &&
    css`
      z-index: -1;
    `}
  ${({theme}) => theme.breakpoints.up('md')} {
    gap: ${({theme}) => theme.spacing(2)};
  }
  ${({theme}) => theme.breakpoints.up('lg')} {
    padding-right: 0;
  }
`

export const NavbarIconButtonWrapper = styled('div')`
  background-color: ${({theme}) => theme.palette.primary.main};
  display: flex;
  justify-content: center;
  align-items: center;
  height: var(--navbar-height);
  aspect-ratio: 1;
  color: ${({theme}) => theme.palette.common.white};
  ${({theme}) => theme.breakpoints.up('md')} {
    svg {
      font-size: ${({theme}) => theme.spacing(4.5)};
    }
  }
  ${({theme}) => theme.breakpoints.up('lg')} {
    svg {
      font-size: ${({theme}) => theme.spacing(6.5)};
    }
  }
`

const useLogoLinkStyles = (isMenuOpen: boolean) => {
  return useMemo(
    () => css`
      ${isMenuOpen &&
      css`
        z-index: -1;
      `}
    `,
    [isMenuOpen]
  )
}

export const NavbarLogoWrapper = styled('div')`
  fill: currentColor;
  width: auto;
`

const useImageStyles = () => {
  const theme = useTheme()
  return useMemo(
    () => css`
      max-height: ${theme.spacing(5)};
      max-width: ${theme.spacing(25)};
      ${theme.breakpoints.up('md')} {
        max-height: ${theme.spacing(6)};
        max-width: ${theme.spacing(30)};
      }
      ${theme.breakpoints.up('lg')} {
        max-height: ${theme.spacing(9)};
        max-width: ${theme.spacing(38)};
      }
    `,
    [theme]
  )
}

type NavbarOpenCloseButtonProps = {
  toggle: UseToggle
}

export const NavbarOpenCloseButton = ({toggle}: NavbarOpenCloseButtonProps) => {
  const {
    elements: {IconButton}
  } = useWebsiteBuilder()
  return (
    <IconButton size="large" aria-label="Menu" onClick={toggle.toggle} color={'inherit'}>
      {!toggle.value && <MdMenu />}
      {toggle.value && <MdClose />}
    </IconButton>
  )
}
