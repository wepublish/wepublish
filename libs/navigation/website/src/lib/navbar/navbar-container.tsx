import {useNavigationListQuery, NavigationListQuery} from '@wepublish/website/api'
import {QueryResult} from '@apollo/client'
import {useEffect} from 'react'
import {useWebsiteBuilder} from '@wepublish/website/builder'

export type NavbarContainerProps = {
  onQuery?: (
    queryResult: Pick<QueryResult<NavigationListQuery>, 'data' | 'loading' | 'error' | 'refetch'>
  ) => void
}

export function NavbarContainer({onQuery}: NavbarContainerProps) {
  const {Navbar} = useWebsiteBuilder()
  const {data, loading, error, refetch} = useNavigationListQuery()

  useEffect(() => {
    onQuery?.({data, loading, error, refetch})
  }, [data, loading, error, refetch, onQuery])

  return <Navbar data={data} loading={loading} error={error} />
}
