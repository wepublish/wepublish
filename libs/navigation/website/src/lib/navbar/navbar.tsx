import styled from '@emotion/styled'
import AppBar from '@mui/material/AppBar'
import Box from '@mui/material/Box'
import Toolbar from '@mui/material/Toolbar'
import Typography from '@mui/material/Typography'
import {BuilderNavbarProps, useWebsiteBuilder} from '@wepublish/website/builder'

const NavbarWrapper = styled(Box)`
  flex-grow: 1;
`

const Title = styled.div`
  flex-grow: 1;
`

export type NavbarProps = BuilderNavbarProps

export function Navbar({className, data, loading, error}: NavbarProps) {
  const {
    elements: {Button}
  } = useWebsiteBuilder()

  return (
    <NavbarWrapper className={className}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" component={Title}>
            Newsroom
          </Typography>

          <Button>Login</Button>
        </Toolbar>
      </AppBar>
    </NavbarWrapper>
  )
}
