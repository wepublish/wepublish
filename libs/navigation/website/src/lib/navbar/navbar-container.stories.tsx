import {action} from '@storybook/addon-actions'
import {Meta} from '@storybook/react'
import {FullNavigationFragment, Navigation, NavigationListDocument} from '@wepublish/website/api'
import {NavbarContainer} from './navbar-container'
import {css} from '@emotion/react'

const navigation = {
  id: 'cldx7kcpi1168oapxftiqsh0p',
  key: 'main',
  name: 'main',
  links: [
    {
      __typename: 'PageNavigationLink',
      label: 'Home',
      page: {
        url: '/'
      }
    },
    {
      __typename: 'ArticleNavigationLink',
      label: 'Artikel',
      article: {
        url: '/a/abcd'
      }
    },
    {
      __typename: 'ExternalNavigationLink',
      label: 'Google',
      url: 'https://google.com'
    }
  ]
} as FullNavigationFragment

const navigations = [
  navigation,
  {
    ...navigation,
    id: '1234-1234',
    key: 'categories',
    name: 'Kategorien'
  },
  {
    ...navigation,
    id: '12345-12345',
    key: 'about',
    name: 'Ãœber Uns'
  }
] as Navigation[]

export default {
  component: NavbarContainer,
  title: 'Container/Navbar'
} as Meta

export const Default = {
  args: {
    onQuery: action('onQuery'),
    slug: 'main',
    categorySlugs: ['categories', 'about']
  },

  parameters: {
    apolloClient: {
      mocks: [
        {
          request: {
            query: NavigationListDocument
          },
          result: {
            data: {
              navigations
            }
          }
        }
      ]
    }
  }
}

export const WithClassName = {
  args: {
    onQuery: action('onQuery'),
    slug: 'main',
    categorySlugs: ['categories', 'about'],
    className: 'extra-classname'
  },

  parameters: {
    apolloClient: {
      mocks: [
        {
          request: {
            query: NavigationListDocument
          },
          result: {
            data: {
              navigations
            }
          }
        }
      ]
    }
  }
}

export const WithEmotion = {
  args: {
    onQuery: action('onQuery'),
    css: css`
      background-color: #eee;
    `,
    slug: 'main',
    categorySlugs: ['categories', 'about']
  },

  parameters: {
    apolloClient: {
      mocks: [
        {
          request: {
            query: NavigationListDocument
          },
          result: {
            data: {
              navigations
            }
          }
        }
      ]
    }
  }
}
