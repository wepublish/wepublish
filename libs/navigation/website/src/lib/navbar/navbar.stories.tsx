import {Navbar, NavbarProps} from './navbar'
import {NavbarContainer} from './navbar-container'
import {action} from '@storybook/addon-actions'
import {ComponentStory, ComponentMeta} from '@storybook/react'
import {NavigationListDocument} from '@wepublish/website/api'

export default {
  component: Navbar,
  title: 'Navbar'
} as ComponentMeta<typeof Navbar>

const Template = (args: NavbarProps) => <Navbar {...args} />
export const Default = Template.bind({})

const ContainerTemplate: ComponentStory<typeof NavbarContainer> = args => (
  <NavbarContainer {...args} />
)
export const WithContainer = ContainerTemplate.bind({})

WithContainer.args = {
  onQuery: action('onQuery')
}

WithContainer.parameters = {
  apolloClient: {
    mocks: [
      {
        request: {
          query: NavigationListDocument
        },
        result: {
          data: {
            navigations: [
              {
                id: 'cldx7kcpi1168oapxftiqsh0p',
                key: 'main',
                name: 'main',
                links: [
                  {
                    __typename: 'PageNavigationLink',
                    label: 'Home',
                    page: {
                      url: '/'
                    }
                  },
                  {
                    __typename: 'ArticleNavigationLink',
                    label: 'Artikel',
                    article: {
                      url: '/a/abcd'
                    }
                  },
                  {
                    __typename: 'ExternalNavigationLink',
                    label: 'Google',
                    url: 'https://google.com'
                  }
                ]
              }
            ]
          }
        }
      }
    ]
  }
}
