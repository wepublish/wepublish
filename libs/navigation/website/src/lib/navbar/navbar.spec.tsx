import {render} from '@testing-library/react'

import {Navbar} from './navbar'

describe('Navbar', () => {
  it('should render successfully', () => {
    const {baseElement} = render(<Navbar data={undefined} loading={true} error={undefined} />)
    expect(baseElement).toBeTruthy()
  })
})
