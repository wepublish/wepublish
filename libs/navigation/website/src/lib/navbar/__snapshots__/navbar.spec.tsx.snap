// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render Default 1`] = `
<DocumentFragment>
  <nav
    class="css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/"
            >
              Home
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/a/abcd"
            >
              Artikel
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://google.com"
            >
              Google
            </a>
          </div>
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;

exports[`Navbar should render WithClassName 1`] = `
<DocumentFragment>
  <nav
    class="extra-classname css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/"
            >
              Home
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/a/abcd"
            >
              Artikel
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://google.com"
            >
              Google
            </a>
          </div>
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;

exports[`Navbar should render WithEmotion 1`] = `
<DocumentFragment>
  <nav
    class="css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/"
            >
              Home
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="/a/abcd"
            >
              Artikel
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
              href="https://google.com"
            >
              Google
            </a>
          </div>
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;

exports[`Navbar should render WithError 1`] = `
<DocumentFragment>
  <nav
    class="css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          />
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;

exports[`Navbar should render WithLoading 1`] = `
<DocumentFragment>
  <nav
    class="css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          />
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;

exports[`Navbar should render WithoutItems 1`] = `
<DocumentFragment>
  <nav
    class="css-4i0nrb"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionStatic css-pivuyh-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ahoq89-MuiToolbar-root"
      >
        <div
          class="css-1hacyix"
        >
          <button
            aria-label="Menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-edgeStart MuiIconButton-sizeLarge css-1snimjj-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <div
            class="css-xqufyo"
          />
        </div>
      </div>
    </header>
  </nav>
</DocumentFragment>
`;
